### PR TITLE
Updated docs w.r.t. infinite doc length changes

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

Updated docs w.r.t. infinite doc length changes. See https://github.com/explosion/spacy-llm/pull/332.

Also updates task list in usage docs with `TranslationTask`.

### Types of change

Docs update.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: website/docs/api/large-language-models.mdx [hunk 1]
```diff
@@ -202,13 +202,82 @@ not require labels.
...
+
+All tasks are registered in the `llm_tasks` registry.
+
+### On Sharding {id="task-sharding"}
+
+"Sharding" describes, generally speaking, the process of distributing parts of a
+dataset across multiple storage units for easier processing and lookups. In
+`spacy-llm` we use this term (synonymously: "mapping") to describe the splitting
+up of prompts if they are too long for a model to handle, and "fusing"
+(synonymously: "reducing") to describe how the model responses for several shars
```
Comment (svlandeg) [inline website/docs/api/large-language-models.mdx:228]:
```suggestion
(synonymously: "reducing") to describe how the model responses for several shards
```

File: website/docs/api/large-language-models.mdx [hunk 2]
```diff
@@ -202,13 +202,82 @@ not require labels.
...
+A toy example: let's assume a model has a context window of 25 tokens and the
+prompt template for our fictional, sharding-supporting task looks like this:
+
+```
+Estimate the sentiment of this text:
+"{text}"
+Estimated entiment:
+```
+
+Depening on how tokens are counted exactly (this is a config setting), we might
```
Comment (svlandeg) [inline website/docs/api/large-language-models.mdx:245]:
```suggestion
Depending on how tokens are counted exactly (this is a config setting), we might
```

File: website/docs/api/large-language-models.mdx [hunk 3]
```diff
@@ -202,13 +202,82 @@ not require labels.
...
+document content however will be split, if the length of the fully rendered
+prompt exceeds a model context length.
+
+A toy example: let's assume a model has a context window of 25 tokens and the
+prompt template for our fictional, sharding-supporting task looks like this:
+
+```
+Estimate the sentiment of this text:
+"{text}"
+Estimated entiment:
```
Comment (svlandeg) [inline website/docs/api/large-language-models.mdx:242]:
```suggestion
Estimated sentiment:
```

File: website/docs/api/large-language-models.mdx [hunk 4]
```diff
@@ -202,13 +202,82 @@ not require labels.
...
+assuming the task used supports sharding, will split the prompt into two (the
+default splitting strategy splits by tokens, but alternative splitting
+strategies splitting e. g. by sentences can be configured):
+
+_(Prompt 1/2)_
+
+```
+Estimate the sentiment of this text:
+"This has been amazing - I can't remember "
+Estimated entiment:
```
Comment (svlandeg) [inline website/docs/api/large-language-models.mdx:262]:
```suggestion
Estimated sentiment:
```

File: website/docs/api/large-language-models.mdx [hunk 5]
```diff
@@ -202,13 +202,82 @@ not require labels.
...
+```
+
+_(Prompt 2/2)_
+
+```
+Estimate the sentiment of this text:
+"the last time I left the cinema so impressed."
+Estimated entiment:
```
Comment (svlandeg) [inline website/docs/api/large-language-models.mdx:270]:
```suggestion
Estimated sentiment:
```

File: website/docs/usage/large-language-models.mdx [hunk 6]
```diff
@@ -340,15 +340,45 @@ A _task_ defines an NLP problem or question, that will be sent to the LLM via a
...
-| [`task.generate_prompts`](/api/large-language-models#task-generate-prompts) | Takes a collection of documents, and returns a collection of "prompts", which can be of type `Any`.                                                          |
-| [`task.parse_responses`](/api/large-language-models#task-parse-responses)   | Takes a collection of LLM responses and the original documents, parses the responses into structured information, and sets the annotations on the documents. |
+Tasks may support prompt sharding (for more info see the API docs on
+[sharding](/api/large-language-models#task-sharding) and
+[non-sharding](/api/large-language-models#task-nonsharding) tasks). The function
+signatures for `generate_prompts` and `parse_responses` depend on whether they
+do.
+
+| _For tasks *not supporting* sharding:_ | Task | Description |     |
+| -------------------------------------- | ---- | ----------- | --- |
```
Comment (svlandeg) [inline website/docs/usage/large-language-models.mdx:355]: Looks like this table format needs to be fixed (cf https://deploy-preview-13214--spacy.netlify.app/usage/large-language-models#tasks)

**Review by svlandeg — COMMENTED** [review]
Thanks, looks good!

Only had a few typo corrections, and two tables that need fixing in terms of formatting.

File: website/docs/usage/large-language-models.mdx (hunk 6 above)
Comment (rmitsch) [reply website/docs/usage/large-language-models.mdx]: Fixed. Dunno how that happened.

**Review by svlandeg — APPROVED** [review]

**Pull request merged**